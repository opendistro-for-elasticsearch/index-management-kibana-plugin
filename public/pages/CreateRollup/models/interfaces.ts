/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { ManagedCatIndex } from "../../../../server/models/interfaces";
import { FieldItem } from "../../../../models/interfaces";

export interface MappingsItem {
  properties?: {};
  type?: string;
}

export interface DimensionItem {
  sequence: number;
  field: FieldItem;
  aggregationMethod: string;
  interval?: number;
}

export interface MetricItem {
  source_field: FieldItem;
  all: boolean;
  min: boolean;
  max: boolean;
  sum: boolean;
  avg: boolean;
  value_count: boolean;
}

//From backend
export interface RollupDimensionItem {
  date_histogram?: {
    fixed_interval?: string;
    calendar_interval?: string;
    source_field: string;
    target_field: string;
    timezone: string;
  };
  terms?: {
    source_field: string;
    target_field: string;
  };
  histogram?: {
    source_field: string;
    target_field: string;
    interval: number;
  };
}

export interface RollupMetricItem {
  source_field: string;
  metrics: [
    {
      min?: Object;
      max?: Object;
      sum?: Object;
      avg?: Object;
      value_count?: Object;
    }
  ];
}
