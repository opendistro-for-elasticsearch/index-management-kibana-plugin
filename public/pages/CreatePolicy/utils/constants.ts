/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

export const DEFAULT_POLICY = JSON.stringify(
  {
    policy: {
      description: "A simple default policy that changes the replica count between hot and cold states.",
      default_state: "hot",
      states: [
        {
          name: "hot",
          actions: [{ replica_count: { number_of_replicas: 5 } }],
          transitions: [{ state_name: "cold", conditions: { min_index_age: "30d" } }],
        },
        {
          name: "cold",
          actions: [{ replica_count: { number_of_replicas: 2 } }],
          transitions: [],
        },
      ],
    },
  },
  null,
  4
);
