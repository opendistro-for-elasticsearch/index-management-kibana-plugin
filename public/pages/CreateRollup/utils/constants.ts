/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

//Added last_updated_time and start_time to avoid error, but they should be system generated.
//Added a delay as 0, but this should be optional, so null should be fine.
export const EMPTY_ROLLUP = JSON.stringify({
  rollup: {
    continuous: false,
    description: "",
    dimensions: [
      {
        date_histogram: {
          source_field: "",
          fixed_interval: "2ms",
          timezone: "Africa/Abidjan",
        },
      },
    ],
    enabled: false,
    metrics: [],
    page_size: 1000,
    roles: [],
    schedule: {
      interval: {
        start_time: 234802,
        period: 1,
        unit: "MINUTES",
      },
    },
    source_index: "",
    target_index: "",
  },
});

export const FixedTimeunitOptions = [
  { value: "ms", text: "Millisecond(s)" },
  { value: "s", text: "Second(s)" },
  { value: "m", text: "Minute(s)" },
  { value: "h", text: "Hour(s)" },
  { value: "d", text: "Day(s)" },
];

export const DelayTimeunitOptions = [
  { value: "SECONDS", text: "Second(s)" },
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const CalendarTimeunitOptions = [
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const AddFieldsColumns = [
  {
    field: "label",
    name: "Field name",
    sortable: true,
  },
  {
    field: "type",
    name: "Field type",
    sortable: true,
    render: (type: string | undefined) => (type == null || type == undefined ? "-" : type),
  },
];
