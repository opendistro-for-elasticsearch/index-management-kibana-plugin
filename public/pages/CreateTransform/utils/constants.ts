/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

export const EMPTY_TRANSFORM = JSON.stringify({
  transform: {
    description: "",
    enabled: true,
    page_size: 1000,
    roles: [],
    schedule: {
      interval: {
        start_time: 234802,
        period: 1,
        unit: "MINUTES",
      },
    },
    source_index: "",
    target_index: "",
  },
});

export const FixedTimeunitOptions = [
  { value: "ms", text: "Millisecond(s)" },
  { value: "s", text: "Second(s)" },
  { value: "m", text: "Minute(s)" },
  { value: "h", text: "Hour(s)" },
  { value: "d", text: "Day(s)" },
];

export const DelayTimeunitOptions = [
  { value: "SECONDS", text: "Second(s)" },
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const CalendarTimeunitOptions = [
  { value: "m", text: "Minute" },
  { value: "h", text: "Hour" },
  { value: "d", text: "Day" },
  { value: "w", text: "Week" },
  { value: "M", text: "Month" },
  { value: "q", text: "Quarter" },
  { value: "y", text: "Year" },
];

export const ScheduleIntervalTimeunitOptions = [
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const AddFieldsColumns = [
  {
    field: "label",
    name: "Field name",
    sortable: true,
  },
  {
    field: "type",
    name: "Field type",
    sortable: true,
    render: (type: string | undefined) => (type == null || type == undefined ? "-" : type),
  },
];

export const ExecutionFrequencyDefinitionOptions = [
  { value: "fixed", text: "Define by fixed interval" },
  { value: "cron", text: "Define by cron expression" },
];

export const DefaultSampleDataSize = 50;
