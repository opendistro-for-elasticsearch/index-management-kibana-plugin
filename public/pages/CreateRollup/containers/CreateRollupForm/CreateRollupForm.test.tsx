/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from "react";
import { fireEvent, render, wait } from "@testing-library/react";
import { Redirect, Route, RouteComponentProps, Switch } from "react-router-dom";
import { MemoryRouter as Router } from "react-router";
import { ServicesConsumer, ServicesContext } from "../../../../services";
import { browserServicesMock } from "../../../../../test/mocks";
import { BrowserServices } from "../../../../models/interfaces";
import { ModalProvider, ModalRoot } from "../../../../components/Modal";
import { BREADCRUMBS, ROUTES } from "../../../../utils/constants";
import CreateRollupForm from "./CreateRollupForm";
import chrome from "ui/chrome";
import userEvent from "@testing-library/user-event";

const sampleMapping = {
  index_1: {
    mappings: {
      properties: {
        category: {
          type: "text",
          fields: {
            keyword: {
              type: "keyword",
            },
          },
        },
        customer_gender: {
          type: "keyword",
        },
        day_of_week: {
          type: "keyword",
        },
        day_of_week_i: {
          type: "integer",
        },
        geoip: {
          properties: {
            city_name: {
              type: "keyword",
            },
            region_name: {
              type: "keyword",
            },
          },
        },
        order_date: {
          type: "date",
        },
        products: {
          properties: {
            _id: {
              type: "text",
              fields: {
                keyword: {
                  type: "keyword",
                  ignore_above: 256,
                },
              },
            },
            category: {
              type: "text",
              fields: {
                keyword: {
                  type: "keyword",
                },
              },
            },
            price: {
              type: "half_float",
            },
            quantity: {
              type: "integer",
            },
            tax_amount: {
              type: "half_float",
            },
            taxful_price: {
              type: "half_float",
            },
            taxless_price: {
              type: "half_float",
            },
          },
        },
        taxful_total_price: {
          type: "half_float",
        },
        taxless_total_price: {
          type: "half_float",
        },
        total_quantity: {
          type: "integer",
        },
        type: {
          type: "keyword",
        },
        user: {
          type: "keyword",
        },
      },
    },
  },
};

const sampleMapping2 = {
  kibana_sample_data_ecommerce: {
    mappings: {
      properties: {
        category: {
          type: "text",
          fields: {
            keyword: {
              type: "keyword",
            },
          },
        },
        currency: {
          type: "keyword",
        },
        customer_birth_date: {
          type: "date",
        },
        customer_first_name: {
          type: "text",
          fields: {
            keyword: {
              type: "keyword",
              ignore_above: 256,
            },
          },
        },
        customer_full_name: {
          type: "text",
          fields: {
            keyword: {
              type: "keyword",
              ignore_above: 256,
            },
          },
        },
        customer_gender: {
          type: "keyword",
        },
        customer_id: {
          type: "keyword",
        },
        customer_last_name: {
          type: "text",
          fields: {
            keyword: {
              type: "keyword",
              ignore_above: 256,
            },
          },
        },
        customer_phone: {
          type: "keyword",
        },
        day_of_week: {
          type: "keyword",
        },
        day_of_week_i: {
          type: "integer",
        },
        email: {
          type: "keyword",
        },
        event: {
          properties: {
            dataset: {
              type: "keyword",
            },
          },
        },
        geoip: {
          properties: {
            city_name: {
              type: "keyword",
            },
            continent_name: {
              type: "keyword",
            },
            country_iso_code: {
              type: "keyword",
            },
            location: {
              type: "geo_point",
            },
            region_name: {
              type: "keyword",
            },
          },
        },
        manufacturer: {
          type: "text",
          fields: {
            keyword: {
              type: "keyword",
            },
          },
        },
        order_date: {
          type: "date",
        },
        order_id: {
          type: "keyword",
        },
        products: {
          properties: {
            _id: {
              type: "text",
              fields: {
                keyword: {
                  type: "keyword",
                  ignore_above: 256,
                },
              },
            },
            base_price: {
              type: "half_float",
            },
            base_unit_price: {
              type: "half_float",
            },
            category: {
              type: "text",
              fields: {
                keyword: {
                  type: "keyword",
                },
              },
            },
            created_on: {
              type: "date",
            },
            discount_amount: {
              type: "half_float",
            },
            discount_percentage: {
              type: "half_float",
            },
            manufacturer: {
              type: "text",
              fields: {
                keyword: {
                  type: "keyword",
                },
              },
            },
            min_price: {
              type: "half_float",
            },
            price: {
              type: "half_float",
            },
            product_id: {
              type: "long",
            },
            product_name: {
              type: "text",
              fields: {
                keyword: {
                  type: "keyword",
                },
              },
              analyzer: "english",
            },
            quantity: {
              type: "integer",
            },
            sku: {
              type: "keyword",
            },
            tax_amount: {
              type: "half_float",
            },
            taxful_price: {
              type: "half_float",
            },
            taxless_price: {
              type: "half_float",
            },
            unit_discount_amount: {
              type: "half_float",
            },
          },
        },
        sku: {
          type: "keyword",
        },
        taxful_total_price: {
          type: "half_float",
        },
        taxless_total_price: {
          type: "half_float",
        },
        total_quantity: {
          type: "integer",
        },
        total_unique_products: {
          type: "integer",
        },
        type: {
          type: "keyword",
        },
        user: {
          type: "keyword",
        },
      },
    },
  },
};

function renderCreateRollupFormWithRouter() {
  return {
    ...render(
      <Router>
        <ServicesContext.Provider value={browserServicesMock}>
          <ServicesConsumer>
            {(services: BrowserServices | null) =>
              services && (
                <ModalProvider>
                  <ModalRoot services={services} />
                  <Switch>
                    <Route
                      path={ROUTES.CREATE_ROLLUP}
                      render={(props: RouteComponentProps) => (
                        <div style={{ padding: "25px 25px" }}>
                          <CreateRollupForm {...props} rollupService={services.rollupService} indexService={services.indexService} />
                        </div>
                      )}
                    />
                    <Route path={ROUTES.ROLLUPS} render={(props) => <div>Testing rollup landing page</div>} />
                    <Redirect from="/" to={ROUTES.CREATE_ROLLUP} />
                  </Switch>
                </ModalProvider>
              )
            }
          </ServicesConsumer>
        </ServicesContext.Provider>
      </Router>
    ),
  };
}

describe("<CreateRollupForm /> spec", () => {
  it("renders the component", async () => {
    const { container } = renderCreateRollupFormWithRouter();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("sets breadcrumbs when mounting", async () => {
    renderCreateRollupFormWithRouter();

    expect(chrome.breadcrumbs.set).toHaveBeenCalledTimes(4);
    expect(chrome.breadcrumbs.set).toHaveBeenCalledWith([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.ROLLUPS, BREADCRUMBS.CREATE_ROLLUP]);
  });

  it("routes back to rollup landing page if cancelled", async () => {
    const { getByTestId, getByText } = renderCreateRollupFormWithRouter();

    expect(getByTestId("createRollupCancelButton")).toBeEnabled();

    userEvent.click(getByTestId("createRollupCancelButton"));

    await wait(() => getByText("Testing rollup landing page"));
  });

  it("will show error on step 1", async () => {
    const { getByTestId, queryByText } = renderCreateRollupFormWithRouter();

    expect(getByTestId("createRollupNextButton")).toBeEnabled();

    userEvent.click(getByTestId("createRollupNextButton"));

    expect(queryByText("Job name is required.")).not.toBeNull();

    expect(queryByText("Source index is required.")).not.toBeNull();

    expect(queryByText("Target index is required.")).not.toBeNull();
  });

  it("routes from step 1 to step 2", async () => {
    const indices = [
      {
        "docs.count": 5,
        "docs.deleted": 2,
        health: "green",
        index: "index_1",
        pri: "1",
        "pri.store.size": "100KB",
        rep: "0",
        status: "open",
        "store.size": "100KB",
        uuid: "some_uuid",
      },
    ];

    browserServicesMock.indexService.getIndices = jest.fn().mockResolvedValue({
      ok: true,
      response: { indices, totalIndices: 1 },
    });

    browserServicesMock.rollupService.getMappings = jest.fn().mockResolvedValue({
      ok: true,
      response: sampleMapping,
    });
    const { getByTestId, getByLabelText, queryByText, getAllByTestId } = renderCreateRollupFormWithRouter();

    fireEvent.focus(getByLabelText("Name"));
    await userEvent.type(getByLabelText("Name"), "some_rollup_id");
    fireEvent.blur(getByLabelText("Name"));

    fireEvent.focus(getByTestId("description"));
    await userEvent.type(getByTestId("description"), "some description");
    fireEvent.blur(getByTestId("description"));

    await userEvent.type(getAllByTestId("comboBoxSearchInput")[0], "index_1");
    fireEvent.keyDown(getAllByTestId("comboBoxSearchInput")[0], { key: "Enter", code: "Enter" });

    await userEvent.type(getAllByTestId("comboBoxSearchInput")[1], "some_target_index");
    fireEvent.keyDown(getAllByTestId("comboBoxSearchInput")[1], { key: "Enter", code: "Enter" });

    userEvent.click(getByTestId("createRollupNextButton"));

    expect(queryByText("Job name is required.")).toBeNull();

    expect(queryByText("Source index is required.")).toBeNull();

    expect(queryByText("Target index is required.")).toBeNull();

    //Check that it routes to step 2
    expect(queryByText("Timestamp field")).not.toBeNull();
  });

  it("routes from step 1 to step 3", async () => {
    const indices = [
      {
        "docs.count": 5,
        "docs.deleted": 2,
        health: "green",
        index: "index_1",
        pri: "1",
        "pri.store.size": "100KB",
        rep: "0",
        status: "open",
        "store.size": "100KB",
        uuid: "some_uuid",
      },
    ];

    browserServicesMock.indexService.getIndices = jest.fn().mockResolvedValue({
      ok: true,
      response: { indices, totalIndices: 1 },
    });

    browserServicesMock.rollupService.getMappings = jest.fn().mockResolvedValue({
      ok: true,
      response: sampleMapping,
    });
    const {
      debug,
      getByTestId,
      getByLabelText,
      queryByText,
      getAllByTestId,
      getByDisplayValue,
      getByText,
    } = renderCreateRollupFormWithRouter();

    fireEvent.focus(getByLabelText("Name"));
    await userEvent.type(getByLabelText("Name"), "some_rollup_id");
    fireEvent.blur(getByLabelText("Name"));

    await userEvent.type(getAllByTestId("comboBoxSearchInput")[0], "index_1");
    fireEvent.keyDown(getAllByTestId("comboBoxSearchInput")[0], { key: "Enter", code: "Enter" });

    await userEvent.type(getAllByTestId("comboBoxSearchInput")[1], "some_target_index");
    fireEvent.keyDown(getAllByTestId("comboBoxSearchInput")[1], { key: "Enter", code: "Enter" });

    userEvent.click(getByTestId("createRollupNextButton"));

    expect(queryByText("Job name is required.")).toBeNull();

    expect(queryByText("Source index is required.")).toBeNull();

    expect(queryByText("Target index is required.")).toBeNull();

    //Check that it routes to step 2
    expect(queryByText("Timestamp field")).not.toBeNull();

    //Select timestamp
    await userEvent.type(getByTestId("comboBoxSearchInput"), "order_date");
    await wait();
    fireEvent.keyDown(getByTestId("comboBoxSearchInput"), { key: "Enter", code: "Enter" });

    //Test calendar interval
    userEvent.click(getByLabelText("Calendar"));
    expect(queryByText("Every 1")).not.toBeNull();

    //Test change interval
    userEvent.click(getByDisplayValue("Hour"));
    await wait(() => getByText("Week"));
    userEvent.click(getByText("Week"));

    //Test change timezone
    userEvent.click(getByLabelText("Timezone"));
    await wait(() => getByText("America/Los_Angeles"));
    userEvent.click(getByText("America/Los_Angeles"));

    //Test add aggregation
    // userEvent.click(getByTestId("addFieldsAggregation"));

    //Make sure modal shows up, check if need to wait
    // await wait(() => getByTestId("addFieldsAggregationCancel"));
    // userEvent.click(getByText("Add"));
    // debug();
    // userEvent.click(getByTestId("checkboxSelectRow-day_of_week_i"));
    // userEvent.click(getByTestId("addFieldsAggregationCancel"));
    // expect(queryByText("keyword")).not.toBeNull();

    userEvent.click(getByTestId("createRollupNextButton"));

    expect(queryByText("Timestamp is required.")).toBeNull();

    //Check that it routes to step 3
    expect(queryByText("Enable job by default")).not.toBeNull();
  });
});
