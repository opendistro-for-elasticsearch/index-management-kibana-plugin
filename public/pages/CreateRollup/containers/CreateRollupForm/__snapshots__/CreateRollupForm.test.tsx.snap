// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateRollupForm /> spec renders the component 1`] = `
<div
  style="padding: 25px 25px;"
>
  <form>
    <div
      style="padding: 5px 50px;"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
          style="max-width: 300px;"
        >
          <div
            style="padding-left: 10px;"
          >
            <div
              class="euiSteps"
            >
              <div
                class="euiStep"
              >
                <div
                  class="euiStep__titleWrapper"
                >
                  <div
                    aria-label="Step 1"
                    class="euiStepNumber euiStep__circle"
                  >
                    1
                  </div>
                  <h6
                    class="euiTitle euiTitle--small euiStep__title"
                  >
                    Set up indices
                  </h6>
                </div>
                <div
                  class="euiStep__content"
                />
              </div>
              <div
                class="euiStep"
              >
                <div
                  class="euiStep__titleWrapper"
                >
                  <div
                    aria-label="Step 2"
                    class="euiStepNumber euiStepNumber--disabled euiStep__circle"
                  >
                    2
                  </div>
                  <h6
                    class="euiTitle euiTitle--small euiStep__title"
                  >
                    Define aggregations and metrics
                  </h6>
                </div>
                <div
                  class="euiStep__content"
                />
              </div>
              <div
                class="euiStep"
              >
                <div
                  class="euiStep__titleWrapper"
                >
                  <div
                    aria-label="Step 3"
                    class="euiStepNumber euiStepNumber--disabled euiStep__circle"
                  >
                    3
                  </div>
                  <h6
                    class="euiTitle euiTitle--small euiStep__title"
                  >
                    Specify schedule
                  </h6>
                </div>
                <div
                  class="euiStep__content"
                />
              </div>
              <div
                class="euiStep"
              >
                <div
                  class="euiStep__titleWrapper"
                >
                  <div
                    aria-label="Step 4"
                    class="euiStepNumber euiStepNumber--disabled euiStep__circle"
                  >
                    4
                  </div>
                  <h6
                    class="euiTitle euiTitle--small euiStep__title"
                  >
                    Review and create
                  </h6>
                </div>
                <div
                  class="euiStep__content"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <h1
            class="euiTitle euiTitle--large"
          >
            Set up indices
          </h1>
          <div
            class="euiSpacer euiSpacer--l"
          />
          <div
            class="euiPanel euiPanel--paddingMedium"
            style="padding-left: 0px; padding-right: 0px;"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
              style="padding: 0px 10px;"
            >
              <div
                class="euiFlexItem"
              >
                <h3
                  class="euiTitle euiTitle--medium"
                >
                  Job name and description
                </h3>
              </div>
            </div>
            <hr
              class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
            />
            <div
              style="padding: initial;"
            >
              <div
                style="padding-left: 10px;"
              >
                <div
                  class="euiSpacer euiSpacer--s"
                />
                <div
                  class="euiFormRow"
                  id="some_html_id-row"
                >
                  <div
                    class="euiFormRow__labelWrapper"
                  >
                    <label
                      aria-invalid="false"
                      class="euiFormLabel euiFormRow__label"
                      for="some_html_id"
                    >
                      Name
                    </label>
                  </div>
                  <div
                    class="euiFormRow__fieldWrapper"
                  >
                    <div
                      class="euiFormControlLayout"
                    >
                      <div
                        class="euiFormControlLayout__childrenWrapper"
                      >
                        <input
                          aria-describedby="some_html_id-help"
                          class="euiFieldText"
                          id="some_html_id"
                          placeholder="my-rollupjob1"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="euiFormHelpText euiFormRow__text"
                      id="some_html_id-help"
                    >
                      Specify a unique, descriptive name.
                    </div>
                  </div>
                </div>
                <div
                  class="euiSpacer euiSpacer--l"
                />
                <div
                  class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--directionRow euiFlexGroup--responsive"
                >
                  <div
                    class="euiFlexItem euiFlexItem--flexGrowZero"
                  >
                    <div
                      class="euiText euiText--extraSmall"
                    >
                      <h4>
                        Description
                      </h4>
                    </div>
                  </div>
                  <div
                    class="euiFlexItem"
                  >
                    <div
                      class="euiText euiText--extraSmall"
                    >
                      <div
                        class="euiTextColor euiTextColor--subdued"
                      >
                        <i>
                           - optional
                        </i>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="euiSpacer euiSpacer--xs"
                />
                <div
                  class="euiFormRow"
                  id="some_html_id-row"
                >
                  <div
                    class="euiFormRow__fieldWrapper"
                  >
                    <textarea
                      class="euiTextArea euiTextArea--resizeVertical euiTextArea--compressed"
                      data-test-subj="description"
                      id="some_html_id"
                      rows="3"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiSpacer euiSpacer--l"
          />
          <div
            class="euiPanel euiPanel--paddingMedium"
            style="padding-left: 0px; padding-right: 0px;"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
              style="padding: 0px 10px;"
            >
              <div
                class="euiFlexItem"
              >
                <h3
                  class="euiTitle euiTitle--medium"
                >
                  Indices
                </h3>
              </div>
            </div>
            <hr
              class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
            />
            <div
              style="padding: initial;"
            >
              <div
                style="padding-left: 10px;"
              >
                <div
                  class="euiSpacer euiSpacer--s"
                />
                <div
                  class="euiCallOut euiCallOut--warning"
                >
                  <div
                    class="euiText euiText--small"
                  >
                    <p>
                      You can't change indices after creating a job. Double-check the source and target index names before proceeding.
                    </p>
                  </div>
                </div>
                <div
                  class="euiSpacer euiSpacer--m"
                />
                <div
                  class="euiFormRow"
                  id="some_html_id-row"
                >
                  <div
                    class="euiFormRow__labelWrapper"
                  >
                    <label
                      aria-invalid="false"
                      class="euiFormLabel euiFormRow__label"
                      for="some_html_id"
                    >
                      Source index
                    </label>
                  </div>
                  <div
                    class="euiFormRow__fieldWrapper"
                  >
                    <div
                      aria-describedby="some_html_id-help"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      class="euiComboBox"
                      data-test-subj="sourceIndexCombobox"
                      role="combobox"
                    >
                      <div
                        class="euiFormControlLayout"
                      >
                        <div
                          class="euiFormControlLayout__childrenWrapper"
                        >
                          <div
                            class="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap euiComboBox__inputWrap-isClearable"
                            data-test-subj="comboBoxInput"
                            tabindex="-1"
                          >
                            <p
                              class="euiComboBoxPlaceholder"
                            >
                              Select source index
                            </p>
                            <div
                              class="euiComboBox__input"
                              style="font-size: 14px; display: inline-block;"
                            >
                              <input
                                aria-controls=""
                                data-test-subj="comboBoxSearchInput"
                                id="some_html_id"
                                role="textbox"
                                style="box-sizing: content-box; width: 2px;"
                                value=""
                              />
                              <div
                                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                              />
                            </div>
                          </div>
                          <div
                            class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                          >
                            <button
                              aria-label="Open list of options"
                              class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                              data-test-subj="comboBoxToggleListButton"
                              type="button"
                            >
                              EuiIconMock
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="euiFormHelpText euiFormRow__text"
                      id="some_html_id-help"
                    >
                      The index pattern on which to performed the rollup job. You can use * as a wildcard.
                    </div>
                  </div>
                </div>
                <div
                  class="euiFormRow"
                  id="some_html_id-row"
                >
                  <div
                    class="euiFormRow__labelWrapper"
                  >
                    <label
                      aria-invalid="false"
                      class="euiFormLabel euiFormRow__label"
                      for="some_html_id"
                    >
                      Target index
                    </label>
                  </div>
                  <div
                    class="euiFormRow__fieldWrapper"
                  >
                    <div
                      aria-describedby="some_html_id-help"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      class="euiComboBox"
                      data-test-subj="targetIndexCombobox"
                      role="combobox"
                    >
                      <div
                        class="euiFormControlLayout"
                      >
                        <div
                          class="euiFormControlLayout__childrenWrapper"
                        >
                          <div
                            class="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap euiComboBox__inputWrap-isClearable"
                            data-test-subj="comboBoxInput"
                            tabindex="-1"
                          >
                            <p
                              class="euiComboBoxPlaceholder"
                            >
                              Select or create target index
                            </p>
                            <div
                              class="euiComboBox__input"
                              style="font-size: 14px; display: inline-block;"
                            >
                              <input
                                aria-controls=""
                                data-test-subj="comboBoxSearchInput"
                                id="some_html_id"
                                role="textbox"
                                style="box-sizing: content-box; width: 2px;"
                                value=""
                              />
                              <div
                                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                              />
                            </div>
                          </div>
                          <div
                            class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                          >
                            <button
                              aria-label="Open list of options"
                              class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                              data-test-subj="comboBoxToggleListButton"
                              type="button"
                            >
                              EuiIconMock
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="euiFormHelpText euiFormRow__text"
                      id="some_html_id-help"
                    >
                      The index stores rollup results. You can look up or an existing index to reuse or type to create a new index.
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
    </div>
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 5px 50px;"
    >
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <button
          class="euiButtonEmpty euiButtonEmpty--primary"
          data-test-subj="createRollupCancelButton"
          type="button"
        >
          <span
            class="euiButtonEmpty__content"
          >
            <span
              class="euiButtonEmpty__text"
            >
              Cancel
            </span>
          </span>
        </button>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <button
          class="euiButton euiButton--primary euiButton--fill"
          data-test-subj="createRollupNextButton"
          type="button"
        >
          <span
            class="euiButton__content"
          >
            <span
              class="euiButton__text"
            >
              Next
            </span>
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;
